ExampleMod/Content/Tiles/ExampleBlock.cs

```diff
@@ -27,5 +30,60 @@ public override void NumDust(int i, int j, bool fail, ref int num) {
 		public override void ChangeWaterfallStyle(ref int style) {
 			style = ModContent.GetInstance<ExampleWaterfallStyle>().Slot;
 		}
+
+		public override bool DrawCracks(SpriteBatch spriteBatch, int i, int j, int damage, int crackStyle, bool isAnimation, int animationTimeElapsed) {
+			Texture2D customCrackTexture = ModContent.Request<Texture2D>("ExampleMod/Content/Tiles/ExampleBlock_Cracks").Value;
+
+			int frame = damage switch {
+				>= 80 => 3,
+				>= 60 => 2,
+				>= 40 => 1,
+				>= 20 => 0,
+				_ => 0
+			};
+
+			int crackFrameHeight = customCrackTexture.Height / 4;
+			Rectangle crackSourceRect = new(0, frame * crackFrameHeight, 16, 16);
+
+            Color lightingColor = Lighting.GetColor(i, j);
+
+			if (isAnimation) {
+				Vector2 baseDrawPos = new Vector2(i * 16 - (int)Main.screenPosition.X, j * 16 - (int)Main.screenPosition.Y);
+				Vector2 drawOrigin = new Vector2(8f, 8f);
+				Vector2 finalDrawPosition = baseDrawPos + drawOrigin;
+
+				spriteBatch.Draw(
+					customCrackTexture,
+					finalDrawPosition,
+					crackSourceRect,
+					Color.White,
+					0f,
+					drawOrigin,
+					1.0f,
+					SpriteEffects.None,
+					0f
+				);
+			} else {
+				Vector2 drawOffset = new Vector2(Main.offScreenRange, Main.offScreenRange);
+				if (Main.drawToScreen) {
+					drawOffset = Vector2.Zero;
+				}
+				Vector2 baseDrawPos = new Vector2(i * 16 - (int)Main.screenPosition.X, j * 16 - (int)Main.screenPosition.Y) + drawOffset;
```

Comment:
having to hardcode these calculations is a pain. We should just pass in a parameter for the draw pos of the tile center in screen space, and then the example can just use `origin: new Vector2(8f, 8f)` regardless. Ideally we can remove the `isAnimation` argument entirely.